### PR TITLE
remove use of deprecated whisk object

### LINKED ISSUE #98: whisk object deprecated
the actions in this package use the now deprecated whisk object - migration to promises (and npm openwhisk) should be in order.

### PR DESCRIPTION
**Pull request by bjustin-ibm**
There are two uses of of the (soon-to-be-removed) whisk object: `whisk.error()` and `whisk.getAuthKey()`

`whisk.error()` is replaced with promise rejection, and `whisk.getAuthKey()` is replaced with the `__OW_API_KEY` environment variable. Promise rejection required a little bit of refactoring to make it work nicely.

Resolves #98

### CODE REVIEW

File: action/messageHubFeed.js [hunk 1]
```diff
@@ -52,25 +42,38 @@ function main(params) {
...
+            };
 
+            doRequest(options)
+                .then(resolve)
+                .catch(reject);
+        }
+    });
+
+    return promise;
+}
```
Comment (dubeejw) [inline action/messageHubFeed.js]: Need a new line after this bracket and before the next function.

File: action/kafkaFeed.js [hunk 2]
```diff
@@ -8,60 +8,68 @@ var request = require('request');
...
-            }
-        };
+        var triggerComponents = params.triggerName.split("/");
+        var namespace = encodeURIComponent(process.env['__OW_NAMESPACE']);
+        var trigger = encodeURIComponent(triggerComponents[2]);
+        var feedServiceURL = 'http://' + params.package_endpoint + '/triggers/' + namespace + '/' + trigger;
+
+        if (params.lifecycleEvent === 'CREATE') {
+            // makes a PUT request to create the trigger in the feed service provider
+            var putTrigger = function(validatedParams) {
```
Comment (dubeejw) [inline action/kafkaFeed.js]: It may be cleaner to break this out into its own function outside of main().

File: action/messageHubFeed.js [hunk 3]
```diff
@@ -10,36 +10,26 @@ var request = require('request');
...
-            .then(function() {
-                console.log("Successfully authenticated with Message Hub");
+        var triggerComponents = params.triggerName.split("/");
+        var namespace = encodeURIComponent(process.env['__OW_NAMESPACE']);
+        var trigger = encodeURIComponent(triggerComponents[2]);
+        var feedServiceURL = 'http://' + params.package_endpoint + '/triggers/' + namespace + '/' + trigger;
 
+        if (params.lifecycleEvent === 'CREATE') {
+            // makes a PUT request to create the trigger in the feed service provider
+            var putTrigger = function(validatedParams) {
```
Comment (dubeejw) [inline action/messageHubFeed.js]: It may be cleaner to break this out into its own function outside of main().

Comment (bjustin-ibm) [inline action/messageHubFeed.js]: I kind of agree. Inner functions are weird and confusing.

That being said, this function isn't used anywhere else _and_ it captures the `triggerURL` and `namespace` vars declared above.

Comment (dubeejw) [inline action/messageHubFeed.js]: You could pass those variables in as arguments?

**Pull request merged**